systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java

```diff
@@ -0,0 +1,329 @@
+import io.strimzi.systemtest.kafkaclients.internalClients.KafkaClientsBuilder;
+import io.strimzi.systemtest.resources.crd.KafkaResource;
+import io.strimzi.systemtest.security.SystemTestCertHolder;
+import io.strimzi.systemtest.security.SystemTestCertManager;
+import io.strimzi.systemtest.storage.TestStorage;
+import io.strimzi.systemtest.templates.crd.KafkaTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaUserTemplates;
+import io.strimzi.systemtest.utils.ClientUtils;
+import io.strimzi.systemtest.utils.RollingUpdateUtils;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaUtils;
+import io.strimzi.systemtest.utils.kubeUtils.controllers.DeploymentUtils;
+import io.strimzi.systemtest.utils.kubeUtils.objects.PodUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+
+/**
+ * Provides test cases to verify custom key-pair (public key + private key) manipulation. For instance:
+ *  1. Replacing `cluster` key-pair (f.e., ca.crt and ca.key) to invoke renewal process
+ *  2. Replacing `clients` key-pair (f.e., user.crt and user.key) to invoke renewal process
+ *
+ * {@link https://strimzi.io/docs/operators/in-development/configuring.html#installing-your-own-ca-certificates-str}
+ * {@link https://strimzi.io/docs/operators/in-development/configuring.html#proc-replacing-your-own-private-keys-str}
+ */
+@ParallelSuite
+public class CustomCaST extends AbstractST {
+
+    private static final Logger LOGGER = LogManager.getLogger(CustomCaST.class);
+
+    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(CustomCaST.class.getSimpleName()).stream().findFirst().get();
+
+    @ParallelNamespaceTest
+    void testReplacingCustomClusterKeyPairToInvokeRenewalProcess(ExtensionContext extensionContext) throws IOException, NoSuchAlgorithmException, InvalidKeySpecException {
+        final TestStorage testStorage = new TestStorage(extensionContext);
+
+        this.replaceCustomCaToInvokeRenewalProcess(extensionContext,
+            // 0. Generate root and intermediate certificate authority with clients CA
+            new SystemTestCertHolder(extensionContext,
+                "CN=" + extensionContext.getRequiredTestClass().getSimpleName() + "ClusterCA",
+                KafkaResources.clusterCaCertificateSecretName(testStorage.getClusterName()),
+                KafkaResources.clusterCaKeySecretName(testStorage.getClusterName())),
```

Comment:
Can be this added to some variable and then pass it to the `replaceCustomCaToInvokeRenewalProcess` function?
TBH I was little bit confused when I went through these tests and found the way how you wrote it. I don't have any problem with that, just if we would have some things like this in variables or something, it would be maybe more clear and it would increase the readability?

How I wrote, I'm not against it, but I was a bit confused 😄

Reply:
I have re-write them :)

Reply:
documenting these tests properly is a key :-) 
Thanks.